staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -244,3 +261,46 @@ func maybeTruncateUserAgent(req *http.Request) string {
 
 	return ua
 }
+
+// copyWithoutManagedFields will make a deep copy of the specified object and will discard
+// managed fields from the copy.
+// The specified object is expected to be a meta.Object or a "list". The specified object obj
+// is treated as readonly and hence not mutated.
+// On return, an error is set if the function runs into any error while removing the managed fields.
+// On return, the boolean value is true if the copy has been made successfully, otherwise false.
+func copyWithoutManagedFields(obj runtime.Object) (runtime.Object, bool, error) {
+	removeManagedFields := func(obj runtime.Object) error {
+		accessor, err := meta.Accessor(obj)
+		if err != nil {
+			return err
+		}
+		accessor.SetManagedFields(nil)
+		return nil
+	}
+
+	isAccessor := true
+	if _, err := meta.Accessor(obj); err != nil {
+		isAccessor = false
+	}
+	isList := meta.IsListType(obj)
+
+	if !isAccessor && !isList {
+		return nil, false, nil
+	}
+
+	copy := obj.DeepCopyObject()
+
+	if isAccessor {
+		if err := removeManagedFields(copy); err != nil {
+			return nil, false, err
+		}
+	}
+
+	if isList {
+		if err := meta.EachListItem(copy, removeManagedFields); err != nil {
+			return nil, false, err
+		}
+	}
+
```

Comment:
`Table` is not covered here.

Reply:
put a todo

Reply:
Does a table have managedfields at all? I don't think.

Reply:
Yes, I see `managedFields` in the elements under `rows`